client/packages/invoices/src/InboundShipment/ListView/AppBarButtons.tsx

```diff
@@ -78,6 +76,17 @@ export const AppBarButtons = ({
     );
   };
 
+  const handleRowClick = async (row: NameRowFragment): Promise<void> => {
+    invoiceModalController.toggleOff();
+    const data = await fetchInternalOrders(row.id);
```

Comment:
Refetching ensures data is up to date, and waits before executing logic based on this data.